src/Propel/Generator/Command/InitCommand.php

```diff
@@ -264,7 +264,7 @@ private function reverseEngineerSchema(OutputInterface $output, array $options)
         $outputDir = sys_get_temp_dir();
 
         $this->getApplication()->setAutoExit(false);
-        $fullDsn = sprintf('%s;user=%s;password=%s', $options['dsn'], urlencode($options['user']), urlencode($options['password']));
+        $fullDsn = sprintf('%s;user=%s;password=%s', $options['dsn'], $options['user'], $options['password']);
```

Comment:
@marcj The `urlencode` function breaks authentication like this...  If I go through `propel init` using the password `H7{"Qj1n>P` The following command
`echo urlencode('H7{"Qj1n>P');` results in `H7%7B%22Qj1n%3EP`
which will not authenticate to a password to a database with the password `H7{"Qj1n>P`

If this is on purpose, then this only allows us to use the set of three special characters `.-_` when using propel init, and this should included in the documentation.  Would you like me to update the documentation?

@staabm We already use a non-encoded password in \Propel\Generator\Command\InitCommand.php::testConnection.
See https://github.com/propelorm/Propel2/blob/master/src/Propel/Generator/Command/InitCommand.php#L92
and
https://github.com/propelorm/Propel2/blob/master/src/Propel/Generator/Command/InitCommand.php#L95

Should I change the code to encoding that as well?  I will look into other encoding functions besides urlencode, since urlencode does not allow authentication with most special characters.  If you have any suggestions, please suggest.